src/pages/basic-concepts/styling.md

```diff
@@ -1,4 +1,510 @@
+
+**This approach is good for…**
+
+- Developers wanting to try out something new in regard to styling
+- Developers that would appreciate some strong opinions in CSS architecture
+
+**This approach is **NOT** good for…**
+
+- Developers that already have strong opinions on styling, and want to control everything themselves
+
+Read on if you’re looking for some strong opinions 🙂. We’ll describe the approach by enforcing a few key rules that should govern how you set your styles:
+
+### Hybrid Scoped + Utility CSS
+
+#### Scoped styles
+
+You don’t need an explanation on component-based design. You already know that reusing components is a good idea. And it’s this idea that got people used to concepts like [Styled Components][styled-components] and [Styled JSX][styled-jsx]. But rather than burden your users with slow load times of CSS-in-JS, Astro has something better: **built-in scoped styles.**
+
+```jsx
+---
+// src/components/Button.astro -->
+---
+<style lang="scss">
+  /* ✅ Locally scoped! */
+  .btn {
+    padding: 0.5em 1em;
+    border-radius: 3px;
+    font-weight: 700;
+  }
+</style>
+<button type="button" class="btn">
+  <slot></slot>
+</button>
+```
+
+_Note: all the examples here use `lang="scss"` which is a great convenience for nesting, and sharing [colors and variables][sass-use], but it’s entirely optional and you may use normal CSS if you wish._
+
+That `.btn` class is scoped within that component, and won’t leak out. It means that you can **focus on styling and not naming.** Local-first approach fits in very well with Astro’s ESM-powered design, favoring encapsulation and reusability over global scope. While this is a simple example, it should be noted that **this scales incredibly well.** And if you need to share common values between components, [Sass’ module system][sass-use] also gets our recommendation for being easy to use, and a great fit with component-first design.
+
+By contrast, Astro does allow global styles via the `:global()` escape hatch, however, this should be avoided if possible. To illustrate this: say you used your button in a `<Nav />` component, and you wanted to style it differently there. You might be tempted to have something like:
+
+```jsx
+---
+// src/components/Nav.astro
+import Button from './Button.astro';
+---
+
+<style lang="scss">
+  .nav :global(.btn) {
+    /* ❌ This will fight with <Button>’s styles */
+  }
+</style>
+
+<nav class="nav">
+  <Button>Menu</Button>
+</nav>
+```
+
+This is undesirable because now `<Nav>` and `<Button>` fight over what the final button looks like. Now, whenever you edit one, you’ll always have to edit the other, and they are no longer truly isolated as they once were (now coupled by a bidirectional styling dependency). It’s easy to see how this pattern only has to repeated a couple times before being afraid that touching any styles _anywhere_ may break styling in a completely different part of the app (queue `peter-griffin-css-blinds.gif`).
```

Comment:
maybe add a knowyourmeme link?

Reply:
I'll gladly add it if you lead me to the right link 😄

Reply:
I couldn't find one D: this is the best I've got: https://css-tricks.com/can-we-create-a-resize-hack-with-container-queries/#h-can-we-debunk-a-classic-css-meme-with-container-queries